### PR TITLE
Muitiple changes: argument parsing, logging, code style and scenario loading

### PR DESCRIPTION
**Pull request by sumpfralle**
Rework various parts of the code:
* parse commandline arguments via `argparse`
* [linux] use more suitable user config directory
* log message handling via `logging` (even for processes forked by `multiprocessing`)
* code-style compliance via `flake8`
* more robust handling of paths
* handle race condition between server and client process
* adjust yaml loading of packages for scenario parsing

The following details probably need some attention:
* I changed the python module paths in the scenario files from `base.*` to `imperialism_remake.base.*`. This change was required for successful comparisons of enums between the running process and the loaded scenario data (see [be8646606882567cd3bbeb26d82c0535de3b7343]).
* The `single_player_start` functions seems to receive `(str, int)` instead of `(str, str)` (see [e2bed3f5c29bcd6b58e9d8e3fbe5b19e79438dfc]). At least this was reported when clicking through the dialog.

### CODE REVIEW

**Comment by sumpfralle** [comment]
With this state of the code, I manage to do the following actions:
* main window:
    * "imperialism remake" is only partly visible in the lower left corner (the second line is missing)
* preferences:
    * icons for the four categories are missing (question marks are shown)
    * interface language selector is empty (no languages to be selected)
* scenario editor:
    * terrain of a tile cannot be changed (the "set terrain" dialog shows only six colored squares; there seem to be no sensitive areas)
    * scenario can be loaded and is displayed properly (colored tiles, icons for cities, lines for rivers and borders)
* game lobby:
    * icons for the four categories are missing (question marks are shown)
    * scenario can be selected by name
    * nations can be selected on the map preview
    * when accepting a nation, another dialog opens
        * the new dialog is insensitive as long as the scenario selection window is still open (it probably should have closed itself)
        * the map view of the new dialog is completely blank (the zoomed map view in the upper left corner, as well as the main map view on the right)

Thus it feels like I am close to being able to see the main game window :)

**Comment by Trilarion** [comment]
Sorry for absence, will review it this evening.

**Comment by sumpfralle** [comment]
There is no need to worry or hurry - please take your time whenever it suits you.
Have fun!

**Comment by Trilarion** [comment]
I finished looking through the commits of this pull request. It all looks very nice. Thanks. Will merge now.

**Comment by Trilarion** [comment]
Thanks for the nice improvements and changes. I saw that you added some TODO/questions. I try to address them in the near future.

**Comment by sumpfralle** [comment]
Great - thank you for merging!

Regarding [my usage description above](https://github.com/Trilarion/imperialism-remake/pull/4#issuecomment-309253322): did you expect some of the things that failed for me, to work? Should I create some tickets?
Or could you summarize the things/actions/features that you expect to work at the moment? (just for giving experimentally minded people an idea of what they can expect)

**Comment by Trilarion** [comment]
I had a longer period of inactivity, so I don't remember if some of the things worked (I was programming almost exclusively on Windows). Let me check in the next days what works now. I will then come back here and all the things not covered can then surely go into tickets.

I can also summarize the things/actions/features that I expect to work at the moment. Will do it in the next days.

**Comment by Trilarion** [comment]
Btw. you are probably well aware of it, but just for absolute clarity I wanted to mention that this project is open source with a certain license and by contributing your content is now also under this license. If for some reason, you should not be okay with it, please say so. Also, I would take the github account name (sumpfralle) for the credits. Hope this is okay too.

**Comment by sumpfralle** [comment]
I appreciate your choice of the license (GPL3 or later).
The credits are also fine for me.

**Comment by Trilarion** [comment]
I added a short description of the current state of the game, i.e. what the code should do and what it shouldn't do.

See [current state](https://github.com/Trilarion/imperialism-remake/blob/master/documentation/development/state.rst).

**Comment by sumpfralle** [comment]
Thank you for your time. This gives me a good perspective of the currently usable parts.
